### DISCUSSION TITLE
PERUSER removal work around

### DISCUSSION THREAD

**Discussion by polibou** [osiset/laravel-shopify#1187] [Q&A]
Hello,

I've been using this package and it has been great.  With new requirements from Shopify it looks like I need to update from version 11 to now 17.  Yeah its been a while and there are a series of issues I need to figure out.  Moving only up to version 16 doesn't seem to be an option since we need to adhere to the new cookieless session standard to get approved. One of my main issue is the removal of PERUSER.  Is there a work around that I am missing?  How is everyone getting the logged in user now?  Without that, how can we log user actions or restrict certain users in different areas of our app?  

My app manages the warehouse and inventory for our stores.  I need to keep track of who is doing what.

I tried to read the documentation on Oauth and there is very little info on PERUSER from Shopify.  Honestly most of it is over my head.

Is there a way to get the current logged in Shopify user that is using the app?  We could easily get the email and name before.  Wanted to see if there was a better solution than implementing a separate login system into our app.

Thanks for any help.

**Comment by Kyon147** [comment]
Hey @polibou 

You'd need to tweak the app to ask for the permissions in the access grant step before the OAUTH accept screen. https://shopify.dev/apps/auth/oauth/access-modes#online-access 

With this https://shopify.dev/apps/auth/oauth/getting-started#redirect-to-the-grant-screen-using-a-3xx-redirect you can as for "PER_USER" but you will need to set up a way of storing a users access token in a table somewhere and handle the session expiry yourself. 

The "ShopSession" was removed which is what handled the per user auth, [you can see the file here](https://github.com/osiset/laravel-shopify/blob/v11.5.0/src/ShopifyApp/Services/ShopSession.php) which might give you some insight into how to reuse it in v17+ but it will require some work.
